### ISSUE TITLE
Problems with language

### ISSUE THREAD

**Issue by thiagotalma** [kartik-v/yii2-widgets#253]
When we define the language using a variant, the script does not understand.

For example, if you set the language to `pt` works. But `pt-BR` does not.

**Comment by kartik-v** [comment]
Not sure that's correct. What widget are you using and what is your config?

**Comment by thiagotalma** [comment]
I tested on a clean installation of yii2-app-advanced, include the demo code in the view and the line in the composer.

Same problem.

composer.json
```JSON
{
    "name": "yiisoft/yii2-app-advanced",
    "description": "Yii 2 Advanced Project Template",
    "keywords": ["yii2", "framework", "advanced", "project template"],
    "homepage": "http://www.yiiframework.com/",
    "type": "project",
    "license": "BSD-3-Clause",
    "support": {
        "issues": "https://github.com/yiisoft/yii2/issues?state=open",
        "forum": "http://www.yiiframework.com/forum/",
        "wiki": "http://www.yiiframework.com/wiki/",
        "irc": "irc://irc.freenode.net/yii",
        "source": "https://github.com/yiisoft/yii2"
    },
    "minimum-stability": "dev",
    "require": {
        "php": ">=5.4.0",
        "yiisoft/yii2": "*",
        "yiisoft/yii2-bootstrap": "*",
        "yiisoft/yii2-swiftmailer": "*",
        "kartik-v/yii2-widget-select2": "*"
    },
    "require-dev": {
        "yiisoft/yii2-codeception": "*",
        "yiisoft/yii2-debug": "*",
        "yiisoft/yii2-gii": "*",
        "yiisoft/yii2-faker": "*"
    },
    "config": {
        "process-timeout": 1800
    },
    "extra": {
        "asset-installer-paths": {
            "npm-asset-library": "vendor/npm",
            "bower-asset-library": "vendor/bower"
        }
    }
}
```
view
```PHP
<?php
use kartik\select2\Select2;
use yii\web\JsExpression;

$formatJs = <<< 'JS'
var formatRepo = function (repo) {
    if (repo.loading) {
        return repo.text;
    }
    var markup =
'<div class="row">' + 
    '<div class="col-sm-5">' +
        '<img src="' + repo.owner.avatar_url + '" class="img-rounded" style="width:30px" />' +
        '<b style="margin-left:5px">' + repo.full_name + '</b>' + 
    '</div>' +
    '<div class="col-sm-3"><i class="fa fa-code-fork"></i> ' + repo.forks_count + '</div>' +
    '<div class="col-sm-3"><i class="fa fa-star"></i> ' + repo.stargazers_count + '</div>' +
'</div>';
    if (repo.description) {
      markup += '<h5>' + repo.description + '</h5>';
    }
    return '<div style="overflow:hidden;">' + markup + '</div>';
};
var formatRepoSelection = function (repo) {
    return repo.full_name || repo.text;
}
JS;
// Register the formatting script
$this->registerJs($formatJs, $this::POS_HEAD);
// script to parse the results into the format expected by Select2
$resultsJs = <<< JS
function (data, params) {
    params.page = params.page || 1;
    return {
        results: data.items,
        pagination: {
            more: (params.page * 30) < data.total_count
        }
    };
}
JS;
// render your widget
echo Select2::widget([
    'language' => 'pt-BR',
    'name' => 'kv-repo-template',
    'value' => '14719648',
    'initValueText' => 'kartik-v/yii2-widgets',
    'options' => ['placeholder' => 'Search for a repo ...'],
    'pluginOptions' => [
        'allowClear' => true,
        'minimumInputLength' => 1,
        'ajax' => [
            'url' => "https://api.github.com/search/repositories",
            'dataType' => 'json',
            'delay' => 250,
            'data' => new JsExpression('function(params) { return {q:params.term, page: params.page}; }'),
            'results' => new JsExpression($resultsJs),
            'cache' => true
        ],
        'escapeMarkup' => new JsExpression('function (markup) { return markup; }'),
        'templateResult' => new JsExpression('formatRepo'),
        'templateSelection' => new JsExpression('formatRepoSelection'),
    ],
]);
```

view-source (Note that the file was included the "pt-BR", but javascript is set for the "pt")
```JS
<!DOCTYPE html>
<html lang="en-US">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="csrf-param" content="_csrf">
    <meta name="csrf-token" content="bUhtaTlzNFJdKhoZYCNyFCZlAitxBncwAAsFKnI.cBVbHT4KUDQFfw==">
    <title></title>
    <link href="/assets/207bae56/css/bootstrap.css" rel="stylesheet">
<link href="/assets/6dc1059d/css/kv-widgets.css" rel="stylesheet">
<link href="/assets/47f7901b/css/select2.css" rel="stylesheet">
<link href="/assets/47f7901b/css/select2-krajee.css" rel="stylesheet">
<link href="/css/site.css" rel="stylesheet">
<script type="text/javascript">var formatRepo = function (repo) {
    if (repo.loading) {
        return repo.text;
    }
    var markup =
'<div class="row">' + 
    '<div class="col-sm-5">' +
        '<img src="' + repo.owner.avatar_url + '" class="img-rounded" style="width:30px" />' +
        '<b style="margin-left:5px">' + repo.full_name + '</b>' + 
    '</div>' +
    '<div class="col-sm-3"><i class="fa fa-code-fork"></i> ' + repo.forks_count + '</div>' +
    '<div class="col-sm-3"><i class="fa fa-star"></i> ' + repo.stargazers_count + '</div>' +
'</div>';
    if (repo.description) {
      markup += '<h5>' + repo.description + '</h5>';
    }
    return '<div style="overflow:hidden;">' + markup + '</div>';
};
var formatRepoSelection = function (repo) {
    return repo.full_name || repo.text;
}
var select2_44fb5d52 = {"allowClear":true,"minimumInputLength":1,"ajax":{"url":"https://api.github.com/search/repositories","dataType":"json","delay":250,"data":function(params) { return {q:params.term, page: params.page}; },"results":function (data, params) {
    params.page = params.page || 1;
    return {
        results: data.items,
        pagination: {
            more: (params.page * 30) < data.total_count
        }
    };
},"cache":true},"escapeMarkup":function (markup) { return markup; },"templateResult":formatRepo,"templateSelection":formatRepoSelection,"theme":"krajee","placeholder":"Search for a repo ...","language":"pt"};
</script></head>
<body>
        <div class="wrap">
        <nav id="w1" class="navbar-inverse navbar-fixed-top navbar" role="navigation"><div class="container"><div class="navbar-header"><button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#w1-collapse"><span class="sr-only">Toggle navigation</span>
<span class="icon-bar"></span>
<span class="icon-bar"></span>
<span class="icon-bar"></span></button><a class="navbar-brand" href="/index.php">My Company</a></div><div id="w1-collapse" class="collapse navbar-collapse"><ul id="w2" class="navbar-nav navbar-right nav"><li class="active"><a href="/index.php?r=site%2Findex">Home</a></li>
<li><a href="/index.php?r=site%2Fabout">About</a></li>
<li><a href="/index.php?r=site%2Fcontact">Contact</a></li>
<li><a href="/index.php?r=site%2Fsignup">Signup</a></li>
<li><a href="/index.php?r=site%2Flogin">Login</a></li></ul></div></div></nav>
        <div class="container">
                        <div class="kv-plugin-loading loading-w0">&nbsp;</div><div class="input-group input-group-md"><select id="w0" class="form-control" name="kv-repo-template" style="width: 100%; display: none;" data-krajee-select2="select2_44fb5d52">
<option value="">Search for a repo ...</option>
<option value="14719648" selected>kartik-v/yii2-widgets</option>
</select></div>        </div>
    </div>

    <footer class="footer">
        <div class="container">
        <p class="pull-left">&copy; My Company 2015</p>
        <p class="pull-right">Powered by <a href="http://www.yiiframework.com/" rel="external">Yii Framework</a></p>
        </div>
    </footer>

    <script src="/assets/194c3a0d/jquery.js"></script>
<script src="/assets/296171e1/yii.js"></script>
<script src="/assets/47f7901b/js/select2.full.js"></script>
<script src="/assets/47f7901b/js/select2-krajee.js"></script>
<script src="/assets/47f7901b/js/i18n/pt-BR.js"></script>
<script src="/assets/207bae56/js/bootstrap.js"></script>
<script type="text/javascript">jQuery(document).ready(function () {
jQuery.when(jQuery('#w0').select2(select2_44fb5d52)).done(initSelect2Loading('w0', '.select2-container--krajee'));
jQuery('#w0').on('select2:opening', function(event){initSelect2DropStyle('w0', 'kv_close_w0', event);});
jQuery('#w0').on('select2:unselect', function(){window.kv_close_w0 = true;});

});</script></body>
</html>
```

**Comment by kartik-v** [comment]
So you are talking about Select2 widget? Thought you were mentioning about all widgets.

Will check and update.

**Comment by thiagotalma** [comment]
Ooops! I'm sorry.

I clicked on the link "Source" in the demo page of Select2 and came here. I did not realize I was in the wrong place.

**Comment by kartik-v** [comment]
Will be resolved via kartik-v/yii2-widget-select2#32
